### PR TITLE
Sync with grandchild

### PR DESCRIPTION
**Pull request by hqhq**
Without this, it's possible that father process exit with
0 before grandchild exit with error.

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

**Comment by hqhq** [comment]
ping @cyphar .

**Comment by cyphar** [comment]
I'll test this next week (I'm busy with exams and things). It looks okay though.

**Comment by hqhq** [comment]
@cyphar PTAL.

File: libcontainer/nsenter/nsexec.c [hunk 1]
```diff
@@ -516,6 +517,14 @@ void nsexec(void)
 				/* This doesn't need to be global, we're in the parent. */
 				int syncfd = syncpipe[1];
 
+				/*
+				 * Father only return when both child and grandchild are
+				 * ready, so we can receive all possible error codes
+				 * generated by children.
+				 */
+				if (ready == 2)
```
Comment (cyphar) [inline libcontainer/nsenter/nsexec.c]: I'd personally just change the `while` loop so it reads `while (ready < 2)` with basically the same comment. That way we can completely remove `goto out`.

**Comment by hqhq** [comment]
@cyphar Updated.

**Comment by cyphar** [comment]
LGTM

**Comment by hqhq** [comment]
@dqminh @mrunalp @crosbymichael PTAL.

**Comment by dqminh** [comment]
LGTM

**Pull request merged**